lib/common.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:flutter/material.dart';
+
+// Suppose this comes from a package and cannot be known by any
+// libraries we invent (and doesn't know about our libraries).
+mixin ShowAnimationsMixin<T extends StatefulWidget> on State<T> {
+  bool get showAnimations => _showAnimations;
+  bool _showAnimations = true;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _showAnimations = !MediaQuery.of(context).disableAnimations;
```

Comment:
Why use MediaQuery here but in the widget use TickerMode (and ignore the TickerMode.enabled flag)?

Reply:
Forget about it, it's my mistake